pkgs/by-name/fr/freedownloadmanager/package.nix

```diff
@@ -0,0 +1,116 @@
+{
+  lib,
+  stdenv,
+  fetchurl,
+  dpkg,
+  kdePackages,
+  gtk3,
+  atk,
+  cairo,
+  gdk-pixbuf,
+  pango,
+  openssl,
+  icu,
+  mysql80,
+  libdrm,
+  autoPatchelfHook,
+  patchelf,
+  desktop-file-utils,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "freedownloadmanager";
+  version = "6.26.1.6177";
+
+  src = fetchurl {
+    url = "http://debrepo.freedownloadmanager.org/pool/main/f/freedownloadmanager/freedownloadmanager_${finalAttrs.version}_amd64.deb";
+    hash = "sha256-yjyRB/4vnVXB8ZmicZxsrENCCUtp7bRR0YY5x0JtmTg=";
+  };
+
+  nativeBuildInputs = [
+    dpkg
+    autoPatchelfHook
+    patchelf
+    kdePackages.wrapQtAppsHook
+    desktop-file-utils
+  ];
+
+  buildInputs = [
+    kdePackages.qtbase
+    kdePackages.qtdeclarative # QML
+    kdePackages.qtmultimedia
+    kdePackages.qtwayland
+    kdePackages.qt5compat
+
+    # GTK3 dependencies (likely for QPA platform theme like qt6gtk2)
+    gtk3
+    atk
+    cairo
+    gdk-pixbuf
+    pango
+
+    openssl
+    icu
+    mysql80 # Provides libmysqlclient.so for the bundled Qt SQL plugin
+
+    libdrm
+  ];
+
+  preFixup = ''
+    local lib_dir="$out/opt/freedownloadmanager/lib"
+    rm -vf $lib_dir/libQt6*.so.6
+    rm -vf $lib_dir/libicu*.so.*
+    rm -vf $lib_dir/libcrypto.so*
+    rm -vf $lib_dir/libssl.so*
+    rm -vrf $out/opt/freedownloadmanager/qml
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin $out/opt $out/share/applications $out/share/pixmaps
+
+    cp -r opt/freedownloadmanager $out/opt/
+
+    if [ -d usr/share/applications ]; then
+      cp -r usr/share/applications/* $out/share/applications/
+    else
+      echo "ERROR: usr/share/applications directory not found after unpack!"
+      exit 1
+    fi
+
+    ln -s $out/opt/freedownloadmanager/fdm $out/bin/fdm
+    ln -s $out/opt/freedownloadmanager/icon.png $out/share/pixmaps/freedownloadmanager.png
+
+    local desktopFile="$out/share/applications/freedownloadmanager.desktop"
```

Comment:
Same goes for here. If you `cp usr/share/applications/freedownloadmanager.desktop $out/share/applications` and that file doesn't exist, the build fails automatically.